src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferTypedStringBuilderAppendOverloadsTests.cs

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.PreferTypedStringBuilderAppendOverloads,
+    Microsoft.NetCore.Analyzers.Runtime.PreferTypedStringBuilderAppendOverloadsFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.PreferTypedStringBuilderAppendOverloads,
+    Microsoft.NetCore.Analyzers.Runtime.PreferTypedStringBuilderAppendOverloadsFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferTypedStringBuilderAppendOverloadsTests
+    {
+        [Theory]
+        [InlineData("int", true)]
+        [InlineData("Int32", true)]
+        [InlineData("string", true)]
+        [InlineData("String", true)]
+        [InlineData("ulong", true)]
+        [InlineData("object", false)]
+        [InlineData("char[]", false)]
+        [InlineData("DateTime", false)]
+        [InlineData("DayOfWeek", false)]
+        public async Task ArgumentIsToStringMethodCallOnLocal_CSharp(string receiverType, bool diagnosticExcepted)
+        {
+            DiagnosticResult[] diagnostics = diagnosticExcepted ?
```

Comment:
For such a test, using markup should significantly simplify the test. Your input would be strings such as `[|int|]`, `[|string|]`, etc. for diagnostic cases and just `object`, `char[]`, etc. for non-diagnostic cases.

Reply:
That makes sense.  Ok.

Reply:
Wouldn't that markup expect the error at the variable declaration, instead of where ToString is being called?

Reply:
For future PRs where you have both an analyzer and a fixer and the fixer can fix all diagnostics generated by the analyzer, you don't need to write separate analyzer and fixer tests - just writing the fixer tests for all scenarios should suffice. @sharwell has been strongly pushing for this approach, it does make the set of tests more compact.

Reply:
> Wouldn't that markup expect the error at the variable declaration, instead of where ToString is being called?

Actually, yeah, @mavasani, won't that get the span wrong?

Reply:
> Wouldn't that markup expect the error at the variable declaration, instead of where ToString is being called?

Yes, you are correct. You'd still need a separate parameter, but the test would be simpler and avoid magic numbers for diagnostic line/column numbers and spans.

Reply:
> You'd still need a separate parameter, but the test would be simpler and avoid magic numbers for diagnostic line/column numbers and spans.

How would it avoid magic numbers?  What are you proposing the separate argument be?

Reply:
```csharp
        [Theory]
        [InlineData("int", true)]
        [InlineData("Int32", true)]
        [InlineData("string", true)]
        [InlineData("String", true)]
        [InlineData("ulong", true)]
        [InlineData("object", false)]
        [InlineData("char[]", false)]
        [InlineData("DateTime", false)]
        [InlineData("DayOfWeek", false)]
        public async Task ArgumentIsToStringMethodCallOnLocal_CSharp(string receiverType, bool diagnosticExcepted)
        {
            var arg = diagnosticExcepted ? "[|value.ToString()|]" : "value.ToString()";
            await VerifyCS.VerifyAnalyzerAsync($@"
                using System;
                using System.Text;
                class C
                {{
                    public void M()
                    {{
                        {receiverType} value = default;
                        var sb = new StringBuilder();
                        sb.Append({arg});
                        sb.Insert(42, {arg});
                    }}
                }}", diagnostics);
        }
```

Reply:
Ok